runtime/array.c

```diff
@@ -583,7 +584,7 @@ CAMLprim value caml_array_fill(value array,
         caml_darken(NULL, old, NULL);
       }
       if (is_val_young_block)
-        Ref_table_add(&Caml_state->minor_tables->major_ref, fp);
+        Ref_table_add(&Caml_state->minor_tables->major_ref, (value*)fp);
```

Comment:
I think you don't need the `(value *)` cast, since `Ref_table_add` contains a cast already.